### PR TITLE
Add new economic parameters functions

### LINKED ISSUE #826: Create functions for the economic parameters calculations
Create functions for calculating `annualized_cost`, `salvage_value_set`,  `salvage_value`, `weight_for_investment_discounts`, and `weight_for_operation_discounts`. These functions should be called to determine the values so we can reuse them for the investment cost of the storage and flows.

- [ ] Create a separate file to include with the functions
- [ ] Add the weights to the fixed cost values in the objective function (assets, storage, flows)

### PR DESCRIPTION
**Pull request by datejada**
# Pull request details

## Describe the changes made in this pull request

New file, functions, and tests for the economic parameter calculations

## List of related issues or pull requests

Closes #826

## Collaboration confirmation

As a contributor I confirm

-   [ ] I read and followed the instructions in README.dev.md
-   [ ] The documentation is up to date with the changes introduced in this Pull Request (or NA)
-   [ ] Tests are passing
-   [ ] Lint is passing

### CODE REVIEW

**Comment by datejada** [comment]
@abelsiqueira we want to use functions for the economic parameter calculations. This is just the first one, but before continuing, we want your feedback on this so that we start replicating it on the other calculations. Thanks!

File: src/create-model.jl [hunk 1]
```diff
@@ -1161,39 +1161,27 @@ function create_model(
...
-        annualized_cost = Dict(
-            (y, a) =>
-                graph[a].discount_rate / (
-                    (1 + graph[a].discount_rate) *
-                    (1 - 1 / (1 + graph[a].discount_rate)^graph[a].economic_lifetime)
-                ) * graph[a].investment_cost[y] for y in Y for a in Ai[y]
-        )
+        # Calculate the economic parameters
+        discount_rate     = Dict(a => graph[a].discount_rate for y in Y for a in Ai[y])
+        economic_lifetime = Dict(a => graph[a].economic_lifetime for y in Y for a in Ai[y])
```
Comment (abelsiqueira) [inline src/create-model.jl:1166]:
```suggestion
        discount_rate     = Dict(a => graph[a].discount_rate for a in A)
        economic_lifetime = Dict(a => graph[a].economic_lifetime for a in A)
```

File: src/economic-parameters.jl [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+"""
+    calculate_annualized_cost(discount_rate, economic_lifetime, investment_cost, years, investable_assets)
+
+Calculates the annualized cost for each asset, both energy assets and transport assets, in each year using provided discount rates, economic lifetimes, and investment costs.
+
+# Arguments
+- `discount_rate::Dict`: A dictionary where the key is an `asset` or a pair of assets `(asset1, asset2)` for transport assets, and the value is the discount rate.
+- `economic_lifetime::Dict`: A dictionary where the key is an `asset` or a pair of assets `(asset1, asset2)` for transport assets, and the value is the economic lifetime.
+- `investment_cost::Dict`: A dictionary where the key is an tuple `(year, asset)` or `(year, (asset1, asset2))` for transport assets, and the value is the investment cost.
```
Comment (abelsiqueira) [inline src/economic-parameters.jl:11]:
```suggestion
- `investment_cost::Dict`: A dictionary where the key is a tuple `(year, asset)` or `(year, (asset1, asset2))` for transport assets, and the value is the investment cost.
```

File: src/economic-parameters.jl [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+Dict{Tuple{Int64, String}, Float64} with 3 entries:
+  (2021, "asset1") => 123.338
+  (2021, "asset2") => 153.918
+  (2022, "asset1") => 135.671
+```
+
+# Example for transport assets
+
+```jldoctest
+discount_rate = Dict(("asset1","asset2") => 0.05, ("asset3","asset4") => 0.07)
```
Comment (abelsiqueira) [inline src/economic-parameters.jl:53]:
```suggestion
discount_rate = Dict(("asset1", "asset2") => 0.05, ("asset3", "asset4") => 0.07)
```

File: src/economic-parameters.jl [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+economic_lifetime = Dict(("asset1", "asset2") => 10, ("asset3", "asset4") => 15)
+
+investment_cost   = Dict((2021, ("asset1","asset2")) => 1000, (2021, ("asset3","asset4")) => 1500,
+                         (2022, ("asset1","asset2")) => 1100, (2022, ("asset3","asset4")) => 1600)
+years = [2021, 2022]
+
+investable_assets = Dict(2021 => [("asset1","asset2"), ("asset3","asset4")],
+                         2022 => [("asset1","asset2")])
```
Comment (abelsiqueira) [inline src/economic-parameters.jl:62]:
```suggestion
investable_assets = Dict(2021 => [("asset1", "asset2"), ("asset3", "asset4")],
                         2022 => [("asset1", "asset2")])
```

File: src/economic-parameters.jl [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+    discount_rate,
+    economic_lifetime,
+    annualized_cost,
+    years,
+    investable_assets,
+)
+    # Create a dict of the years beyond the last milestone year
+    end_of_horizon = maximum(years)
+    salvage_value_set = Dict(
+        (year, asset) => collect(end_of_horizon+1:year+economic_lifetime[asset]-1) for
```
Comment (abelsiqueira) [inline src/economic-parameters.jl:194]:
```suggestion
        (year, asset) => end_of_horizon+1:year+economic_lifetime[asset]-1 for
```
Maybe `collect` is not necessary

**Pull request merged**